account_invoice_production_lot/models/account_invoice.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+# Copyright 2011 Domsense s.r.l. <http://www.domsense.com>
+# Copyright 2013 Lorenzo Battistini <lorenzo.battistini@agilebg.com>
+# Copyright 2017 Vicent Cubells <vicent.cubells@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountInvoiceLine(models.Model):
+    _inherit = "account.invoice.line"
+
+    order_line_ids = fields.Many2many(
+        comodel_name='sale.order.line',
+        relation='sale_order_line_invoice_rel',
+        column1='invoice_id',
+        column2='order_line_id',
+        string='Order Lines',
+        readonly=True,
+    )
+
+    prod_lot_ids = fields.Many2many(
+        comodel_name='stock.production.lot',
+        compute='_compute_prod_lots',
+        string="Production Lots",
+    )
+
+    lot_formatted_note = fields.Html(
+        string='Formatted Note',
+        compute='_compute_line_lots',
+    )
+
+    @api.multi
+    def _compute_prod_lots(self):
+        for line in self:
+            if not line.move_line_ids and not line.order_line_ids:
+                return
+            if line.move_line_ids:
+                line.prod_lot_ids = self.mapped(
+                    'move_line_ids.lot_ids')
+            else:
+                line.prod_lot_ids = self.mapped(
```

Comment:
Why don't you just simply use this and remove the dependency as stated. There's no other way now of creating invoices (you can't invoice from pickings), so this is enough.

Reply:
Or is there any rough edge I don't find?

Reply:
Ok, I understand.